core/src/main/scala/cats/data/Chain.scala

```diff
@@ -308,13 +310,14 @@ sealed abstract class Chain[+A] extends ChainCompat[A] {
       arg match {
         case Wrap(seq) =>
           if (count == 1) {
-            lhs.append(seq.last)
+            seq.last +: rhs
```

Comment:
this was the bug. The rest of the changes are just optimizations to the code or scalacheck improvements.